packages/system/src/use-style-config.ts

```diff
@@ -72,9 +82,62 @@ export function useStyleConfig(themeKey: any, props: any = {}, opts: any = {}) {
     }
 
     return stylesRef.current
-  }, [styleConfig, mergedProps, opts?.isMultiPart])
+  }, [styleConfig, breakpoints, mergedProps, opts?.isMultiPart])
 }
 
 export function useMultiStyleConfig(themeKey: string, props: any) {
   return useStyleConfig(themeKey, props, { isMultiPart: true })
 }
+
+interface ResolveResponsivePropStylesOptions {
+  responsiveValue: ResponsiveValue<string>
+  responsiveStyles: Record<string, SystemStyleObject>
+  breakpoints: string[]
+  props: Record<string, any>
+}
+
+function resolveResponsivePropStyles({
```

Comment:
I am not 100% satisfied with the `resolveResponsivePropStyles` implementation. Suggestions are welcome!